src/Row.js

```diff
@@ -95,12 +97,17 @@ export default class Row extends Component {
     onPanResponderRelease: (e, gestureState) => {
       if (this._wasLongPress) {
         this._toggleActive(e, gestureState);
-
+      } else if (this._wasPress) {
+        this._cancelLongPress();
+        if(this.props.onPress) {
+          this.props.onPress();
+          this._wasPress = false;
+        }
       } else if (this._isTouchInsideElement(e)) {
         this._cancelLongPress();
```

Comment:
can't understand when it will enter this branch, long press is handled in the first if when this._wasLongPress === true, short press is handled in the second branch when this._wasPress === true

Reply:
It will enter this branch if  this._wasLongPress === false. That means there was a press event but timeout did not worked out, because event was short.

Reply:
It is tested on real ios devices and simulator

Reply:
> It will enter this branch if this._wasLongPress === false. That means there was a press event but timeout did not worked out, because event was short.

that means that this._wasPress === true, so it will enter the  previous else if

Reply:
I'm sorry, I didn't understand earlier what branch do you mean. Yes, you're right.

Reply:
But I don't understand why this branch is needed. You have onLongPress and onPress, what else?

Reply:
```
    onPanResponderRelease: (e, gestureState) => {
      if (this._wasLongPress) {
        this._toggleActive(e, gestureState);
      } else if (this._wasPress && this.props.onPress)) {
        this._cancelLongPress();
        this.props.onPress();
        this._wasPress = false;
      } else if (this._isTouchInsideElement(e)) {
        this._cancelLongPress();

        if (this.props.onLongPress) {
          this.props.onLongPress();
        }
      }
},

```

May be like this

Reply:
also, there is more common question. why we need the new callback in Row? there is onActivate which is called on long press

Reply:
on long press...

Reply:
https://github.com/gitim/react-native-sortable-list/issues/61

Reply:
it does not fixes the issue, because a bug, that was reported occurs when TouchableOpacity placed inside a row and onPress passed to this TouchableOpacity is not fires

Reply:
Ok, I see. In my case it was fixed because my touchable had a size of a whole row.

Reply:
Thanks for replies :)